src/app/components/typography.tsx

```diff
@@ -97,19 +97,35 @@ const titleStyles = (as: Headings) => {
   }
 };
 
-export const Title = forwardRefWithAs<BoxProps, Headings>((props, ref) => (
-  <BaseText
-    userSelect="none"
-    letterSpacing="-0.01em"
-    fontFamily="'Open Sauce One'"
-    fontWeight="500"
-    color={color('text-title')}
-    ref={ref}
-    display="block"
-    css={titleStyles(props.as)}
-    {...props}
-  />
-));
+interface TitleProps extends BoxProps {
+  /** Removes ::before pseudoelement inserted by capsize */
+  removeBeforePseudoElement?: boolean;
+}
+export const Title = forwardRefWithAs<TitleProps, Headings>((props, ref) => {
+  // This is to prevent tall graphemes (like emojis, some chinese characters,
+  // and others) getting cut off by pseudoelements inserted by capsize, which is
+  // due for removal:
+  // https://github.com/hirosystems/stacks-wallet-web/issues/2344
+  const styleObject = props.removeBeforePseudoElement
+    ? { ...titleStyles(props.as), ['::before']: undefined }
+    : titleStyles(props.as);
+  // Avoids passing this prop down to components not expecting it.
+  const { removeBeforePseudoElement: _omit, ...restProps } = props;
```

Comment:
Note on consistency, pattern we often follow is to destructure the props on the first line of the component (search codebase for `= props;`). This way, we get to use the prop directly, and non-style props already omitted:

```suggestion
  const { removeBeforePseudoElement, ...restProps } = props;
  // Prevents tall graphemes (like emojis, some chinese characters,
  // and others) getting cut off by pseudoelements inserted by capsize, which is
  // due for removal:
  // https://github.com/hirosystems/stacks-wallet-web/issues/2344
  const styleObject = removeBeforePseudoElement
    ? { ...titleStyles(props.as), ['::before']: undefined }
    : titleStyles(props.as);
```

Reply:
All good points, will adopt this to improve comments + readability